src/web/builds.rs

```diff
@@ -111,6 +122,102 @@ pub(crate) async fn build_list_json_handler(
         .into_response())
 }
 
+async fn crate_version_exists(
+    mut conn: DbConnection,
+    name: &String,
+    version: &Version,
+) -> Result<bool, anyhow::Error> {
+    let row = sqlx::query!(
+        r#"
+        SELECT COUNT(*) as "count!"
```

Comment:
(no change request) 

for high load situations, this would be slightly less efficient than the previous query since now we have to count :) 

But for the current load it wouldn't matter.

Reply:
> for high load situations, this would be slightly less efficient than the previous query since now we have to count :)

As I mentioned in my reply to your earlier note, I was thinking of that too, but as we presumably should only get a count of 0 or 1, then iff the database batches from 2 rows onward, it will iterate over the full dataset in either case. I don't know for sure whether that "iff" is true though.

Reply:
I could also have used `SELECT 1 FROM ...` (and `LIMIT 1`) to side step the need of disambiguating `id`, and that might have yet another slight edge :). A quick Google didn't bring up anything other for the purpose (there's also `EXISTS` if a query like ours needs to be used as a subquery in another one).